### PR TITLE
Update LeakCanary library

### PR DESCRIPTION
**Pull request by TacoTheDank**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
- Update LeakCanary library 2.9.1 -> 2.12 ([changelog](https://square.github.io/leakcanary/changelog/))
- Fix LeakCanary deprecations

#### APK testing
The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AudricV** [comment]
Version 2.11 has been released yesterday and it fixes some bugs. Could you bump the version update to this version and try to see if everything still works fine?

File: app/build.gradle [hunk 1]
```diff
@@ -290,7 +290,7 @@ dependencies {
     // Memory leak detection
     implementation "com.squareup.leakcanary:leakcanary-object-watcher-android:${leakCanaryVersion}"
     implementation "com.squareup.leakcanary:plumber-android:${leakCanaryVersion}"
```
Comment (Stypox) [inline app/build.gradle:292]: Mmmh, why are these not `debugImplementation`?

**Review by Stypox — APPROVED** [review]

**Comment by TobiGr** [comment]
I get this crash when Android Studio tries to lauch the app after installing it. It's an endless loop.
```
FATAL EXCEPTION: main
                                                                                                    Process: org.schabi.newpipe.debug.bugfixfeedcrash, PID: 6951
                                                                                                    java.lang.RuntimeException: Unable to create application org.schabi.newpipe.DebugApp: java.lang.IllegalStateException: AppWatcher already installed, see exception cause for prior install call
                                                                                                    	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6717)
                                                                                                    	at android.app.ActivityThread.access$1300(ActivityThread.java:237)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1913)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loop(Looper.java:223)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7656)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
                                                                                                    Caused by: java.lang.IllegalStateException: AppWatcher already installed, see exception cause for prior install call
                                                                                                    	at leakcanary.AppWatcher.manualInstall(AppWatcher.kt:98)
                                                                                                    	at leakcanary.AppWatcher.manualInstall$default(AppWatcher.kt:94)
                                                                                                    	at org.schabi.newpipe.DebugApp.onCreate(DebugApp.kt:17)
                                                                                                    	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1192)
                                                                                                    	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6712)
                                                                                                    	at android.app.ActivityThread.access$1300(ActivityThread.java:237) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1913) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                                    	at android.os.Looper.loop(Looper.java:223) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7656) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
                                                                                                    Caused by: java.lang.RuntimeException: manualInstall() first called here
                                                                                                    	at leakcanary.AppWatcher.manualInstall(AppWatcher.kt:116)
                                                                                                    	at leakcanary.AppWatcher.manualInstall$default(AppWatcher.kt:94)
                                                                                                    	at leakcanary.internal.MainProcessAppWatcherInstaller.onCreate(MainProcessAppWatcherInstaller.kt:21)
                                                                                                    	at android.content.ContentProvider.attachInfo(ContentProvider.java:2388)
                                                                                                    	at android.content.ContentProvider.attachInfo(ContentProvider.java:2358)
                                                                                                    	at android.app.ActivityThread.installProvider(ActivityThread.java:7239)
                                                                                                    	at android.app.ActivityThread.installContentProviders(ActivityThread.java:6780)
                                                                                                    	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6697)
                                                                                                    	at android.app.ActivityThread.access$1300(ActivityThread.java:237) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1913) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                                    	at android.os.Looper.loop(Looper.java:223) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7656) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
2023-07-14 15:11:46.152  6951-6951  ACRA                    org...newpipe.debug.bugfixfeedcrash  E  ACRA caught a RuntimeException for org.schabi.newpipe.debug.bugfixfeedcrash
                                                                                                    java.lang.RuntimeException: Unable to create application org.schabi.newpipe.DebugApp: java.lang.IllegalStateException: AppWatcher already installed, see exception cause for prior install call
                                                                                                    	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6717)
                                                                                                    	at android.app.ActivityThread.access$1300(ActivityThread.java:237)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1913)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loop(Looper.java:223)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7656)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
                                                                                                    Caused by: java.lang.IllegalStateException: AppWatcher already installed, see exception cause for prior install call
                                                                                                    	at leakcanary.AppWatcher.manualInstall(AppWatcher.kt:98)
                                                                                                    	at leakcanary.AppWatcher.manualInstall$default(AppWatcher.kt:94)
                                                                                                    	at org.schabi.newpipe.DebugApp.onCreate(DebugApp.kt:17)
                                                                                                    	at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1192)
                                                                                                    	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6712)
                                                                                                    	at android.app.ActivityThread.access$1300(ActivityThread.java:237) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1913) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                                    	at android.os.Looper.loop(Looper.java:223) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:7656) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
                                                                                                    Caused by: java.lang.RuntimeException: manualInstall() first called here
                                                                                                    	at leakcanary.AppWatcher.manualInstall(AppWatcher.kt:116)
                                                                                                    	at leakcanary.AppWatcher.manualInstall$default(AppWatcher.kt:94)
                                                                                                    	at leakcanary.internal.MainProcessAppWatcherInstaller.onCreate(MainProcessAppWatcherInstaller.kt:21)
                                                                                                    	at android.content.ContentProvider.attachInfo(ContentProvider.java:2388)
                                                                                                    	at android.content.ContentProvider.attachInfo(ContentProvider.java:2358)
                                                                                                    	at android.app.ActivityThread.installProvider(ActivityThread.java:7239)
                                                                                                    	at android.app.ActivityThread.installContentProviders(ActivityThread.java:6780)
                                                                                                    	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6697)
```

**Comment by TacoTheDank** [comment]
@TobiGr Fixed. I forgot to disable the auto-install. https://github.com/TeamNewPipe/NewPipe/pull/10231

**Pull request merged**